CONTRIBUTING.md

```diff
@@ -19,15 +31,149 @@ pnpm lint
 pnpm fix
 ```
 
-## Documentation
+### Documentation
 
 Build and publish documentation with:
 
 ```bash
-pnpm doc
-pnpm doc:publish
+pnpm typedoc
+pnpm typedoc:publish
+```
+
+### Publishing
+
+This package follows semantic versioning:
+
+ 1. Major Releases: Introduce significant changes, including breaking changes.
+ 2. Minor Releases: Add new features in a backwards-compatible manner.
+ 3. Patch Releases: Include backwards-compatible bug fixes.
+
+There are multiple inputs that go into deciding when to publish a release, usually driven by external factors like shipping something on the server side and wanting to test it.
+
+Currently, versions are manually updated in the `package.json` file of each package.
+This process involves determining the appropriate version number based on the changes being introduced (e.g., major, minor, patch).
+
+When the package is published to [npm](https://www.npmjs.com/package/@nucypher/taco), tags are used to indicate the environment or network compatibility.
+We use `devnet`, `testnet`, and `mainnet` help users understand which network the release is intended for.
+
+```bash
+npm dist-tag add @nucypher/taco@0.3.0 devnet
+```
+
+Ideally, this would all be automated using Github actions, workflows, releases, and tags.
+
+# External API
+This is the api that we expose to developers.
+It is defined in [`packages/taco/src/taco.ts`](https://github.com/nucypher/taco-web/blob/main/packages/taco/src/taco.ts)
+
+# Design and Architecture
```

Comment:
Is [`CONTRIBUTING.md`](https://contributing.md/how-to-build-contributing-md/) the right place to discuss the design and architecture? Should we move this to Threshold Docs or `taco` package docs?

Reply:
It's possibly the wrong heading - maybe Structure? it's supposed to be a section that helps new contributors get to grips with how the whole repo is laid out and organised

Reply:
If it has to do with repo structure, we should describe how the `pnpm` workspace works, not API design